internal/node/npm_package_bin.bzl

```diff
@@ -8,26 +8,45 @@ _ATTRS = {
     "outs": attr.output_list(),
     "args": attr.string_list(mandatory = True),
     "data": attr.label_list(allow_files = True, aspects = [module_mappings_aspect]),
+    "out_dir": attr.string(),
```

Comment:
should be added to doc string for rule and comment on `$@` usage in args

Reply:
could out_dir be a label to give the user more freedom as to where it will be?

Reply:
I don't think that's possible since https://docs.bazel.build/versions/master/skylark/lib/actions.html#declare_directory is the only API to produce TreeArtifact and it takes string - and you can't use a label to reference something that doesn't exist yet
anyway it's probably good that the output directory has to be inside the package that owns it?

Reply:
👍

Reply:
added missing doc

Also handle the case of `args=["--out-dir=$@"]` (it's a substring of the arg)